charts/pravega/templates/pravega.yaml

```diff
@@ -5,43 +5,106 @@ metadata:
   externalAccess:
     enabled: {{ .Values.externalAccess.enabled }}
+    {{- if .Values.externalAccess.enabled }}
     type: {{ .Values.externalAccess.type }}
-  bookkeeperUri: {{ .Values.bookkeeperUri }}
+    {{- if .Values.externalAccess.domainName }}
+    domainName: {{ .Values.externalAccess.domainName }}
+    {{- end }}
+    {{- end }}
   pravega:
     {{- if .Values.externalAccess.enabled }}
     controllerServiceAccountName: {{ .Values.serviceAccount.name }}
     segmentStoreServiceAccountName: {{ .Values.serviceAccount.name }}
+    {{- if .Values.controller.service.type }}
+    controllerExtServiceType: {{ .Values.controller.service.type }}
+    {{- end }}
+    {{- if .Values.controller.service.annotations }}
+    controllerSvcAnnotations: {{ .Values.controller.service.annotations }}
+    {{- end }}
+    {{- if .Values.segmentStore.service.type }}
+    segmentStoreExtServiceType: {{ .Values.segmentStore.service.type }}
+    {{- end }}
+    {{- if .Values.segmentStore.service.annotations }}
+    segmentStoreSvcAnnotations: {{ .Values.segmentStore.service.annotations }}
+    {{- end }}
     {{- end }}
     image:
-      repository: {{ .Values.pravega.image.repository }}
-    controllerReplicas: {{ .Values.pravega.controllerReplicas }}
+      repository: {{ .Values.image.repository }}
+      pullPolicy: {{ .Values.image.pullPolicy }}
+    controllerReplicas: {{ .Values.controller.replicas }}
     controllerResources:
       requests:
-        memory: "1Gi"
-        cpu: "1000m"
+        memory: {{ .Values.controller.resources.requests.memory | quote }}
+        cpu: {{ .Values.controller.resources.requests.cpu | quote }}
       limits:
-        memory: "3Gi"
-        cpu: "2000m"
-    segmentStoreReplicas: {{ .Values.pravega.segmentStoreReplicas }}
+        memory: {{ .Values.controller.resources.limits.memory | quote }}
+        cpu: {{ .Values.controller.resources.limits.cpu | quote }}
+    segmentStoreReplicas: {{ .Values.segmentStore.replicas }}
     segmentStoreResources:
       requests:
-        memory: "3Gi"
-        cpu: "1000m"
+        memory: {{ .Values.segmentStore.resources.requests.memory | quote }}
+        cpu: {{ .Values.segmentStore.resources.requests.cpu | quote }}
       limits:
-        memory: "5Gi"
-        cpu: "2000m"
-    debugLogging: {{ .Values.pravega.debugLogging }}
+        memory: {{ .Values.segmentStore.resources.limits.memory | quote }}
+        cpu: {{ .Values.segmentStore.resources.limits.cpu | quote }}
+    {{- if .Values.segmentStore.env }}
+    segmentStoreEnvVars: {{ .Values.segmentStore.env }}
+    {{- end }}
+    {{- if .Values.segmentStore.secret }}
+    segmentStoreSecret:
+      {{- if .Values.segmentStore.secret.name }}
+      secret: {{ .Values.segmentStore.name }}
+      {{- end }}
+      {{- if .Values.segmentStore.secret.path }}
+      mountPath: {{ .Values.segmentStore.path }}
+      {{- end }}
+    {{- end }}
+    controllerjvmOptions: {{ .Values.controller.jvmOptions }}
```

Comment:
To keep the consistency I would use `JVM` rather than `jvm` in the name of the properties.

Reply:
I believe the JVM properties are looking for lists, which may be harder to give in a single value like that. We may have to do something like this instead:
```
    controllerJVMOptions:
      {{- if .Values.controller.jvmOptions }}
{{ toYaml .Values.controller.jvmOptions | indent 6 }}
      {{- end }}
```

Thoughts?
(and then similar for the other JVM options)

Reply:
The naming convention for `segmentStoreJVMOptions` and `controllerjvmOptions` is not uniform currently. Changing the name to `controllerJVMOptions` would involve code changes and can be done in a separate PR as that does not come under chart changes.

Reply:
Makes sense for the name! The rest of my comment still stands (about using `toYaml` instead of a range over key/value pairs)